src/sql/run.py

```diff
@@ -358,11 +358,15 @@ def fetchmany(size):
 )
 
 
-def _commit(conn, config):
+def _commit(conn, config, manual_commit):
     """Issues a commit, if appropriate for current config and dialect"""
 
-    _should_commit = config.autocommit and all(
-        dialect not in str(conn.dialect) for dialect in _COMMIT_BLACKLIST_DIALECTS
+    _should_commit = (
+        config.autocommit
+        and all(
+            dialect not in str(conn.dialect) for dialect in _COMMIT_BLACKLIST_DIALECTS
+        )
+        and not manual_commit
```

Comment:
shouldn't this be `and manual_commit`? looks like that controls whether we should or not run the commit command

Reply:
also, let's add a message so the user knows when the commit failed:

so change this:

```python
except sqlalchemy.exc.OperationalError:
    pass # not all engines can commit
```

to something like this:
```python
except sqlalchemy.exc.OperationalError:
    # not all engines can commit
   print("The database does not support the COMMIT command")
```